src/System.Management.Automation/FormatAndOutput/common/ColumnWidthManager.cs

```diff
@@ -107,18 +107,31 @@ private bool AssignColumnWidths(Span<int> columnWidths)
             }
 
             // we still have room and we want to add more width
+            int unfixedColumnCount = 0;
+            foreach (bool columnIsFixed in fixedColumn)
+            {
+                if (!columnIsFixed)
+                {
+                    unfixedColumnCount++;
+                }
+            }
 
-            while (availableWidth > 0)
+            if (unfixedColumnCount > 0)
             {
+                int extraSpace = availableWidth / unfixedColumnCount;
+                int remainder = availableWidth % unfixedColumnCount;
+
                 for (int k = 0; k < columnWidths.Length; k++)
                 {
                     if (fixedColumn[k])
                         continue;
 
-                    columnWidths[k]++;
-                    availableWidth--;
-                    if (availableWidth == 0)
-                        break;
+                    if (remainder > 0 && k <= remainder - 1)
+                    {
+                        columnWidths[k] += extraSpace + 1;
+                        continue;
+                    }
+                    columnWidths[k] += extraSpace;
                 }
```

Comment:
I guess we could make the cycle body more simple and readable like:
```c#
                for (int k = 0; k < columnWidths.Length; k++)
                {
                    if (!fixedColumn[k])
                    {
                        columnWidths[k] += extraSpace;
                        if (remainder-- > 0) columnWidths[k]++;
                    }
                }
```

Reply:
I took this suggestion and implemented it as it's much more readable and simple to follow.